samples/42.scaleout/src/main/java/com/microsoft/bot/sample/scaleout/ScaleoutBot.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.microsoft.bot.sample.scaleout;
+
+import com.microsoft.bot.builder.ActivityHandler;
+import com.microsoft.bot.builder.TurnContext;
+import com.microsoft.bot.dialogs.Dialog;
+
+import java.util.concurrent.CompletableFuture;
+
+/**
+ * Represents a bot that processes incoming Activities.
+ * For each user interaction, an instance of this class is created and the OnTurnAsync method is called.
+ */
+public class ScaleoutBot <T extends Dialog> extends ActivityHandler {
+
+    private final Store store;
+    private final Dialog dialog;
+
+    /**
+     * Initializes a new instance of the {@link ScaleoutBot} class.
+     * @param withStore The store we will be using.
+     * @param withDialog The root dialog to run.
+     */
+    public ScaleoutBot(Store withStore, T withDialog) {
+        if (withStore == null) {
+            throw new IllegalArgumentException("withStore can't be null");
+        }
+        store = withStore;
+
+        if (withDialog == null) {
+            throw new IllegalArgumentException("withDialog can't be null");
+        }
+        dialog = withDialog;
+    }
+
+    @Override
+    protected CompletableFuture<Void> onMessageActivity(TurnContext turnContext) {
+        // Create the storage key for this conversation.
+        String key = String.format(
+            "%s/conversations/%s",
+            turnContext.getActivity().getChannelId(),
+            turnContext.getActivity().getConversation().getId());
```

Comment:
In C# has the `?` which means that there is a chance that it is null, do we have to add a verification?